### PR TITLE
[GraphBolt] enable to invoke gb samplers in functional form

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/feature_fetcher.py [hunk 1]
```diff
@@ -2,9 +2,12 @@
 
 from typing import Dict
 
+from torch.utils.data import functional_datapipe
+
 from torchdata.datapipes.iter import Mapper
 
 
+@functional_datapipe("fetch_feature")
```
Comment (frozenbugs) [inline python/dgl/graphbolt/feature_fetcher.py:10]: feature_fetcher

File: python/dgl/graphbolt/impl/neighbor_sampler.py [hunk 2]
```diff
@@ -1,10 +1,13 @@
 """Neighbor subgraph samplers for GraphBolt."""
 
+from torch.utils.data import functional_datapipe
+
 from ..subgraph_sampler import SubgraphSampler
 from ..utils import unique_and_compact_node_pairs
 from .sampled_subgraph_impl import SampledSubgraphImpl
 
 
+@functional_datapipe("sample_neighbor")
```
Comment (frozenbugs) [inline python/dgl/graphbolt/impl/neighbor_sampler.py:10]: ditto.

Comment (frozenbugs) [reply python/dgl/graphbolt/impl/neighbor_sampler.py:10]: and others, no need to invent new naming convention.

File: tests/python/pytorch/graphbolt/impl/test_negative_sampler.py [hunk 3]
```diff
@@ -5,6 +5,75 @@
...
+            assert len(dst) == batch_size * (negative_ratio + 1)
+            assert len(labels) == batch_size * (negative_ratio + 1)
+            assert torch.all(torch.eq(labels[:batch_size], 1))
+            assert torch.all(torch.eq(labels[batch_size:], 0))
+
+    # Invoke UniformNegativeSampler via class constructor.
+    negative_sampler = gb.UniformNegativeSampler(
+        item_sampler,
+        negative_ratio,
+        gb.LinkPredictionEdgeFormat.INDEPENDENT,
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/impl/test_negative_sampler.py:63]: Conflict with https://github.com/dmlc/dgl/pull/6298, please be careful when merging.

**Review by frozenbugs — APPROVED** [review]

File: python/dgl/graphbolt/impl/neighbor_sampler.py (hunk 2 above)
Comment (Rhett-Ying) [reply python/dgl/graphbolt/impl/neighbor_sampler.py:10]: I am following the naming style of `datapipe`: https://pytorch.org/data/beta/dp_tutorial.html#naming. then will be consistent.

Comment (Rhett-Ying) [reply python/dgl/graphbolt/impl/neighbor_sampler.py:10]: from my perspective, verb looks more reasonable for functional calls.

Comment (frozenbugs) [reply python/dgl/graphbolt/impl/neighbor_sampler.py:10]: I see, if there is already a name convention, let's follow it.

File: tests/python/pytorch/graphbolt/impl/test_negative_sampler.py (hunk 3 above)
Comment (frozenbugs) [reply tests/python/pytorch/graphbolt/impl/test_negative_sampler.py:63]: Feel free to merge, and ask @peizhou001 to clean this up in #6298

**Pull request merged**